src/containers/FactoryLoader.tsx

```diff
@@ -0,0 +1,215 @@
+import { History } from 'history';
+import { delay } from '../services/delay';
+import { AppState } from '../store';
+import * as FactoryResolverStore from '../store/FactoryResolver';
+import * as WorkspaceStore from '../store/Workspaces';
+import FactoryLoaderPage from '../pages/FactoryLoader';
+import { selectAllWorkspaces, selectWorkspaceById } from '../store/Workspaces/selectors';
+import { WorkspaceStatus } from '../services/workspaceStatus';
+
+const WS_ATTRIBUTES_TO_SAVE: string[] = ['workspaceDeploymentLabels', 'workspaceDeploymentAnnotations'];
+
+type Props =
+  MappedProps
+  & { history: History };
+
+export enum LoadFactorySteps {
+  INITIALIZING = 1,
+  LOOKING_FOR_DEVFILE = 2,
+  APPLYING_DEVFILE = 3,
+  CREATE_WORKSPACE = 4,
+  START_WORKSPACE = 5,
+  OPEN_IDE = 6
+}
+
+type State = {
+  search?: string;
+  location?: string;
+  devfileLocationInfo?: string;
+  currentStep: LoadFactorySteps;
+  hasError: boolean;
+};
+
+export class FactoryLoader extends React.PureComponent<Props, State> {
+  private loadFactoryPageCallbacks: { showAlert?: (variant: AlertVariant, title: string) => void } = {};
+  private factoryResolver: FactoryResolverStore.State;
+
+  constructor(props: Props) {
+    super(props);
+
+    const { search } = this.props.history.location;
+
+    this.state = {
+      currentStep: LoadFactorySteps.INITIALIZING,
+      hasError: false,
+      search,
+    };
+  }
+
+  public showErrorAlert(message: string): void {
+    this.setState({ hasError: true });
+    if (this.loadFactoryPageCallbacks.showAlert) {
+      this.loadFactoryPageCallbacks.showAlert(AlertVariant.danger, message);
+    } else {
+      console.error(message);
+    }
+  }
+
+  public componentDidMount(): void {
+    this.createWorkspaceFromFactory();
+  }
+
+  public async componentDidUpdate(): Promise<void> {
+    const { history, workspace, factoryResolver } = this.props;
+    if (this.state.search !== history.location.search) {
+      this.setState({
+        search: history.location.search,
+        hasError: false,
+      });
+      return this.createWorkspaceFromFactory();
+    }
+
+    if (factoryResolver) {
+      this.factoryResolver = factoryResolver;
+    }
+
+    if (this.state.currentStep === LoadFactorySteps.START_WORKSPACE && workspace && WorkspaceStatus[workspace.status] === WorkspaceStatus.RUNNING) {
+      this.setState({ currentStep: LoadFactorySteps.OPEN_IDE });
+      await delay();
+      history.push(`/ide/${workspace.namespace}/${workspace.devfile.metadata.name}`);
+    }
+
+    if (workspace && WorkspaceStatus[workspace.status] === WorkspaceStatus.ERROR) {
+      this.setState({ hasError: true });
+    }
+  }
+
+  private async createWorkspaceFromFactory(): Promise<void> {
+    const { search } = this.props.history.location;
+    if (this.props.workspace) {
+      this.props.clearWorkspaceId();
+    }
+    if (!search) {
+      this.showErrorAlert('Failed to find search params.');
```

Comment:
When this can happen? 
1. User open load-factory without repository?
2. search params are lost after oauth flow?

I'm just curious that it tells nothing to user and it's not clear what should be done to fix the issue.
Should we instead tell something like `Repository/Devfile URL is missing. Please specify it via url query param: $CHE_HOST/dashboard/next#load-factory?url= ` ?

Reply:
I have done
![Screenshot from 2020-11-09 15-13-40](https://user-images.githubusercontent.com/6310786/98546594-9af7af00-229f-11eb-8739-8bbc03994df6.png)

Reply:
`File devfile.yaml is not found in the repository root. Default environment will be applied` does not seem to be consistent with the error message.
Could it be changed to something like `Devfile is not found` ?

Reply:
Ah, I see that `Devfile is not found ...` is place holder but that's not good to show such a message before we checked the URL... Like in the following place, after checking we changed our mind and say that devfile now is found.
![Screenshot_20201109_153832](https://user-images.githubusercontent.com/5887312/98548312-03479000-22a2-11eb-9715-5e70fdd77ca5.png)
![Screenshot_20201109_154103](https://user-images.githubusercontent.com/5887312/98548316-03e02680-22a2-11eb-88bf-8c7ab8948539.png)

Reply:
![Screenshot from 2020-11-10 15-57-09](https://user-images.githubusercontent.com/6310786/98690866-16c42b00-2376-11eb-91c7-0aea8cdbce06.png)